src/cata_tiles.cpp

```diff
@@ -1560,75 +1560,189 @@ bool cata_tiles::draw_from_id_string( std::string id, tripoint pos, int subtile,
                                             ll, apply_night_vision_goggles, height_3d );
 }
 
-bool cata_tiles::draw_from_id_string(std::string id, TILE_CATEGORY category,
-                                     const std::string &subcategory, tripoint pos,
-                                     int subtile, int rota, lit_level ll,
-                                     bool apply_night_vision_goggles, int &height_3d )
+const tile_type *cata_tiles::find_furniture_looks_like( std::string id )
+{
+    std::string looks_like = id;
+    int cnt = 0;
+    while( !looks_like.empty() && cnt < 10 ) {
+        const tile_type *lltt = find_tile_with_season( looks_like );
+        if( lltt ) {
+            return lltt;
+        }
+        const furn_str_id fid( looks_like );
+        if( !fid.is_valid() ) {
+            return nullptr;
+        }
+        const furn_t &furn = fid.obj();
+        looks_like = furn.looks_like;
+        cnt += 1;
+    }
+    return nullptr;
+}
+
+const tile_type *cata_tiles::find_terrain_looks_like( std::string id )
```

Comment:
These functions looks very alike. They introduce unnecessary duplication which can be avoided with something like:

```c++
const tile_type *cata_tiles::find_tile_looks_like( const std::string& id, TILE_CATEGORY category )
{
    switch( category )
    {
	...
	
	case C_FURNITURE: {
	    const furn_str_id fid( id );
	    return fid ? fid->looks_like : nullptr;	
	}
	
	...	
    }
    
    return nullptr;
}

```

Reply:
Yeah, I realized that after I submitted it. Refactoring it is on the todo list.